### PR TITLE
Split jtag fix

### PR DESCRIPTION
**Pull request by That-Canadian**
@drashna Fixed that function issue, sorry about that. Let's let the travis build finish this time =P

### CODE REVIEW

**Comment by drashna** [comment]
Already got a fix for this, actually. https://github.com/qmk/qmk_firmware/pull/4310

But whichever is cleaner would be best. :)

**Comment by That-Canadian** [comment]
@drashna up to you, I just moved the function to the top of the file haha. Moving the function definition is #4310 could be seen as cleaner =P I don't mind either way haha.

**Comment by drashna** [comment]
Just thinking about this. `disable_JTAG` should be called in `keyboard_slave_setup`, not the scan loop.

**Comment by That-Canadian** [comment]
@drashna it's actually called right before it goes into the while(1) loop, so it's only called once.

**Comment by That-Canadian** [comment]
But yeah, moving it to the keyboard_slave_setup() would make a bit more sense

**Comment by drashna** [comment]
Ah, okay, that should be fine then.

**Comment by drashna** [comment]
Also, if we do want this version in here, you may want to remove this line: 
https://github.com/qmk/qmk_firmware/blob/109fe0c3b956db93e3d7ae0e18286b748bf201dd/quantum/split_common/split_util.c#L105

Otherwise, we could probably just close this, for now.

**Comment by vomindoraan** [comment]
Without going too deeply into this, I suggest one of two options:

1. If `disable_JTAG()` isn't needed anywhere outside of this file, mark the function `static` so that this intent is made clear. (In this case, if you're defining it above where it's used, then remove the forward declaration line, yeah.)
2. Otherwise, move the declaration to a common header, where other parts of code that might need it can find it.

File: quantum/split_common/split_util.c [hunk 1]
```diff
@@ -32,6 +32,16 @@ volatile bool isLeftHand = true;
 
 volatile uint8_t setTries = 0;
 
+// Temporary code to disable JTAG on the slave board
+void disable_JTAG(void) {
```
Comment (drashna) [inline quantum/split_common/split_util.c]:
```suggestion
static void disable_JTAG(void) {
```
@vomindoraan you mean like this?

Comment (vomindoraan) [reply quantum/split_common/split_util.c]: Yep, exactly!

Comment (drashna) [reply quantum/split_common/split_util.c]: @That-Canadian This sound good?

**Comment by vomindoraan** [comment]
This functionality is also used [here](https://github.com/qmk/qmk_firmware/blob/master/tmk_core/common/keyboard.c#L154-L158), and possibly in a few other places, so maybe use this PR as an opportunity to make it a proper library function?

Put the declaration in [`tmk_core/common/keyboard.h`](https://github.com/qmk/qmk_firmware/blob/dc6c587cab469d141dcfa65fe409954b8df13e7e/tmk_core/common/keyboard.h#L68) and put the definition in [`tmk_core/common/keyboard.c`](https://github.com/qmk/qmk_firmware/blob/dc6c587cab469d141dcfa65fe409954b8df13e7e/tmk_core/common/keyboard.c#L317). Afterwards, replace the [lines I linked](https://github.com/qmk/qmk_firmware/blob/master/tmk_core/common/keyboard.c#L154-L158) in `keyboard.c` with a call to the function.

(While you're at it, I suggest renaming the function to lowercase `disable_jtag()`.)

**Comment by drashna** [comment]
Actually, I'm going to close this for now, since #4328 supersedes this.

**Pull request closed without merging**